### PR TITLE
Add wazuh agent to red team and bastion

### LINKED ISSUE #46: add wazuh agent to all other systems in the detonationlab
add wazuh agent to all other systems in the detonationlab

### PR DESCRIPTION
**Pull request by sonofagl1tch**
closes #46 

- add wazuh agent to all other systems in the detonationlab
- added conditional statement to register wazuh agent with a different name depending on it's internal IP address
- updated IP conditionals for more systems

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: additionalInstallationScripts/installWazuhAgent.sh [hunk 1]
```diff
@@ -21,7 +21,24 @@ yum install wazuh-agent -y -q -e 0
...
+then
+  agentName="redTeam"
+elif [ "$hostIP" == "172.16.0.21" ]
+then
+  agentName="wazuhServer"
+else
+  agentName="bastion"
+fi
+  
+until /var/ossec/bin/agent-auth -m $MANAGER_IP -A $agentName; do
```
Comment (mgmacias95) [inline additionalInstallationScripts/installWazuhAgent.sh]: According to the [`agent-auth` documentation](https://documentation.wazuh.com/current/user-manual/reference/tools/agent-auth.html), `hostname` will be used as agent name if no other is specified under `-A` option. I think it's more elegant and maintainable to configure instances hostname in the template and remove `-A` option from here 😄 .

As a possible way to do it, I found this [stackoverflow thread](https://stackoverflow.com/questions/40915572/setting-aws-ec2-instance-hostname-using-cloudformation-template).

File: awsDetonationLab.template [hunk 2]
```diff
@@ -2080,6 +2088,22 @@
           "Ref": "VPC"
         },
         "SecurityGroupIngress": [
+          {
+            "IpProtocol": "udp",
```
Comment (mgmacias95) [inline awsDetonationLab.template:2099]: Will this be necessary after merging #38?

**Review by mgmacias95 — CHANGES_REQUESTED** [review]

File: additionalInstallationScripts/installWazuhAgent.sh (hunk 1 above)
Comment (sonofagl1tch) [reply additionalInstallationScripts/installWazuhAgent.sh]: made this change added code to set the instance hostname on first boot in commit https://github.com/sonofagl1tch/AWSDetonationLab/pull/47/commits/c7596c513f3d0dd32239747d9a29a13f35345047

Comment (mgmacias95) [reply additionalInstallationScripts/installWazuhAgent.sh]: Hello @sonofagl1tch,

Once that change is added, you can remove the if condition to set `agentName` variable. In fact, that isn't working: all linux agents are registered using `bastion` name, leading to the following:
```shellsession
# cat /var/ossec/etc/client.keys
002 windowsVictim 172.16.0.23 216d1f13c8b66a39ecc89010eec9c555122ab128e9dd96dfa6434ab6dbd9bbfe
004 bastion 172.16.0.20 580b432501d9ddb731dc828a1d41dda58c7b69d63f33e92e4bd1bb28279e7919
```
Only one linux agent is registered.

Comment (mgmacias95) [reply additionalInstallationScripts/installWazuhAgent.sh]: Fixed in 73080b91092ec8c55c9a8a14d7da326910eabc1e
```shellsession
# /var/ossec/bin/agent_control -l

Wazuh agent_control. List of available agents:
   ID: 000, Name: wazuhServer (server), IP: 127.0.0.1, Active/Local
   ID: 001, Name: linuxVictim, IP: 172.16.0.22, Active
   ID: 002, Name: windowsVictim, IP: 172.16.0.23, Active
   ID: 003, Name: bastion, IP: 172.16.0.11, Active
   ID: 004, Name: redTeam, IP: 172.16.0.20, Active
```

**Review by mgmacias95 — APPROVED** [review]
LGTM!

**Pull request merged**